crates/driver/src/blockchain/mod.rs

```diff
@@ -1,30 +1,46 @@
 use {
     crate::logic::eth,
-    ethcontract::{transport::DynTransport, Web3},
     thiserror::Error,
     url::Url,
+    web3::{Transport, Web3},
 };
 
 pub mod contracts;
 
 /// The Ethereum blockchain.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct Ethereum {
-    web3: Web3<DynTransport>,
+    web3: Web3<web3::transports::Http>,
     chain_id: eth::ChainId,
+    network_id: eth::NetworkId,
 }
 
 impl Ethereum {
     /// Access the Ethereum blockchain through an RPC API hosted at the given
     /// URL.
-    pub fn eth_rpc(_url: Url) -> Self {
-        todo!()
+    pub async fn ethrpc(url: &Url) -> Result<Self, web3::Error> {
+        // TODO Enable batching, reuse ethrpc? Put it in the boundary module?
+        // I feel like what we have in shared::ethrpc could be simplified if we use
+        // web3::transports::batch or something, but I haven't looked deep into it, just
+        // a gut feeling.
```

Comment:
Metrics, logging and auto-batching are the three reasons we use it. It is basically a custom `Transport` implementation that we use. It can live alone in a `ethrpc` crate AFAIU.